node-repository/src/main/java/com/yahoo/vespa/hosted/provision/restapi/v2/filter/NodeIdentifier.java

```diff
@@ -31,9 +37,23 @@
     private final Zone zone;
     private final NodeRepository nodeRepository;
 
+    private final AtomicReference<List<Node>> nodes;
+    private static final Logger logger = Logger.getLogger(NodeIdentifier.class.getName());
+
     NodeIdentifier(Zone zone, NodeRepository nodeRepository) {
         this.zone = zone;
         this.nodeRepository = nodeRepository;
+        nodes = new AtomicReference<>(nodeRepository.getNodes());
+        new ScheduledThreadPoolExecutor(1, ThreadFactoryFactory.getDaemonThreadFactory("node-identifier-refresh"))
```

Comment:
Executor should be shut down when deconstructing. Consider using a Guava supplier, that way you won't have to manage an executor / thread pool.

E.g.
```
Supplier<List<Node>> nodeCache = Suppliers.memoizeWithExpiration(nodeRepository::getNodes, 1, TimeUnit.MINUTES);
```

`nodeCache.get()` will return the cached value for 1 minute.

Reply:
Why? this will only run in the config server and should only shut down when shutting down the config server

Reply:
True, still think it's cleaner to avoid starting new threads while it's shutting down though. 🙂